pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -941,19 +969,187 @@ func TestReconcile_CustomTask(t *testing.T) {
 				t.Errorf("Expected reason %q but was %s", v1beta1.PipelineRunReasonRunning.String(), condition.Reason)
 			}
 
-			if len(reconciledRun.Status.Runs) != 1 {
-				t.Errorf("Expected PipelineRun status to include one Run status, got %d", len(reconciledRun.Status.Runs))
-			}
-			if _, exists := reconciledRun.Status.Runs[tc.wantRun.Name]; !exists {
-				t.Errorf("Expected PipelineRun status to include Run status but was %v", reconciledRun.Status.Runs)
-			}
+			verifyRunStatusesCount(t, config.DefaultEmbeddedStatus, reconciledRun.Status, 1)
+			verifyRunStatusesNames(t, config.DefaultEmbeddedStatus, reconciledRun.Status, tc.wantRun.Name)
+		})
+	}
+}
+
+// TestReconcile_CustomTaskWithEmbeddedStatus runs "Reconcile" on a PipelineRun
+// with one Custom Task reference that has not been run yet, and each possible
+// value for the "embedded-status" feature flag. It verifies that the Run is
+// created, it checks the resulting API actions, status and events.
+func TestReconcile_CustomTaskWithEmbeddedStatus(t *testing.T) {
+	testCases := []struct {
+		name              string
+		embeddedStatusVal string
+	}{
+		{
+			name:              "default embedded status",
+			embeddedStatusVal: config.DefaultEmbeddedStatus,
+		},
+		{
+			name:              "full embedded status",
+			embeddedStatusVal: config.FullEmbeddedStatus,
+		},
+		{
+			name:              "both embedded status",
+			embeddedStatusVal: config.BothEmbeddedStatus,
+		},
+		{
+			name:              "minimal embedded status",
+			embeddedStatusVal: config.MinimalEmbeddedStatus,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			runTestReconcileCustomTaskWithEmbeddedStatus(t, tc.embeddedStatusVal)
 		})
 	}
 }
 
+func runTestReconcileCustomTaskWithEmbeddedStatus(t *testing.T, embeddedStatus string) {
```

Comment:
it looks like this function may accidentally have been duplicated with the one above?

Reply:
Ah, yeah, I think I added this one while initially prototyping, before the table-based changes. I'm gonna remove this.

Reply:
I can't tell where that links to, but I think the pattern you have going of `func TestReconcile(t *testing.T)` +  `func runTestReconcileWithEmbeddedStatus(t *testing.T, embeddedStatus string)` works well, and it would also be nice to use it here for consistency. can you explain a bit more about the concern with doing it that way?

Reply:
Ah, yeah, you have to expand the diff to see it.

So `TestReconcile_CustomTask` is already a table-based test - I didn't want to do the matrix-y thing of running each of its testcases once for each `embedded-status` value, so instead I added two new test cases to `TestReconcile_CustomTask`, using the same input `PipelineRun`/expected `Run` as the existing "custom task with taskref" test case, but with `embedded-status=full` and `embedded-status=minimal`. So it doesn't fit exactly the same pattern as the `runTest...` ones, but I think it's a better fit for this particular test.

Reply:
Honestly, in practice, we could probably skip having test cases in `TestReconcile_CustomTask` for different `embedded-status` values, since we test reconciliation of a `Run` with the different `embedded-status` values in `runTestUpdatePipelineRunStatusFromInformer` now anyway.